mypy tests busted

I'm not sure if anyone uses these except @Callek , so I'm tempted to turn them off.
However, we have the work done, and it seems like other people like strict typing in python, so that's an argument for fixing them if time allows.

Bustage in https://firefoxci.taskcluster-artifacts.net/RZTJNzBoSUuxhH1nlc9ENQ/0/public/logs/live_backing.log, 

```
[0m[1mmypy installed: aiohttp==3.7.4.post0,aiomemoizettl==0.0.3,arrow==1.0.3,async-timeout==3.0.1,attrs==20.3.0,certifi==2020.12.5,cffi==1.14.5,chardet==4.0.0,cryptography==3.4.7,dictdiffer==0.8.1,github3.py==2.0.0,idna==2.10,immutabledict==1.3.0,importlib-metadata==3.10.1,json-e==4.4.1,jsonschema==3.2.0,jwcrypto==0.8,mohawk==1.1.0,multidict==5.1.0,mypy==0.812,mypy-extensions==0.4.3,pycparser==2.20,pyrsistent==0.17.3,python-dateutil==2.8.1,PyYAML==5.4.1,requests==2.25.1,-e git+https://github.com/escapewindow/scriptworker.git@0772efb48c15555c0588ab586cf87cd32ce9897a#egg=scriptworker,six==1.15.0,slugid==2.0.0,taskcluster==39.2.0,taskcluster-urls==13.0.1,typed-ast==1.4.3,typing-extensions==3.7.4.3,uritemplate==3.0.1,urllib3==1.26.4,yarl==1.6.3,zipp==3.4.1
[0m[1mmypy run-test-pre: PYTHONHASHSEED='3092831942'
[0m[1mmypy run-test: commands[0] | mypy --config /scriptworker/mypi.ini /scriptworker/src
[0msrc/scriptworker/constants.py:34: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:34: error: Need type annotation for 'DEFAULT_CONFIG'
src/scriptworker/constants.py:42: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:52: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:81: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:94: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:101: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:112: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:114: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:130: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:132: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:168: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:170: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:208: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:211: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:226: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:235: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:257: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:266: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:271: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:274: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:283: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:292: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:304: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:346: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:362: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:364: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:385: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:397: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:421: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:427: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:428: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:429: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:432: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:433: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:444: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:446: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:487: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:503: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:513: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:514: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:515: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:516: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:517: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:518: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:531: error: Call to untyped function "immutabledict" in typed context
src/scriptworker/constants.py:547: error: Call to untyped function "immutabledict" in typed context
Found 47 errors in 1 file (checked 18 source files)
```

--- a/mypi.ini
+++ b/mypi.ini
@@ -21,6 +21,8 @@ disallow_untyped_defs = True
 [mypy-scriptworker.constants]
 check_untyped_defs = True
 disallow_untyped_defs = True
+# untyped_calls because of immutabledict==1.3.0
+disallow_untyped_calls = False
 
 [mypy-scriptworker.context]
 check_untyped_defs = True

--- a/src/scriptworker/constants.py
+++ b/src/scriptworker/constants.py
@@ -31,7 +31,7 @@ STATUSES = {
 #
 # When adding new complex config, make sure all `list`s are `tuple`s, and all
 # `dict`s are `immutabledict`s!  (This should get caught by config tests.)
-DEFAULT_CONFIG = immutabledict(
+DEFAULT_CONFIG: immutabledict[str, Any] = immutabledict(
     {
         "taskcluster_root_url": os.environ.get("TASKCLUSTER_ROOT_URL", "https://firefox-ci-tc.services.mozilla.com/"),
         # Worker identification

